community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreUpgrader.java

```diff
@@ -116,8 +118,6 @@ public void migrateIfNeeded( File storeDirectory)
             throw new UpgradeNotAllowedByConfigurationException();
         }
 
-        CustomIOConfigValidator.assertCustomIOConfigNotUsed( config, CUSTOM_IO_EXCEPTION_MESSAGE );
-
```

Comment:
@burqen I meant checking against specific versions, importing `StandardV` classes, etc. `StoreUpgrader` is currently free of such logic, while `StoreMigrator` is full of it already. `StoreMigrater` is also the only `Migrator` implementation that is incompatible with custom IO configurations.